### DISCUSSION TITLE
Export secret files on demand

### DISCUSSION THREAD

**Discussion by infogulch** [twpayne/chezmoi#3757] [Ideas]
Hey I thought it would be interesting if chezmoi could be set up to export file secrets on demand. For example, when git or ssh goes to open your key file chezmoi would call the password manager to export the file data, render it as a template if applicable, and return it to the opener. It would use FUSE I guess. This would be nice because you could secure the entire machine by just locking the password manager.

**Comment by twpayne** [comment]
Interesting idea. This would, however, require chezmoi to run as a daemon and execute template files on demand, which chezmoi currently does not do. There is further complexity in that FUSE mounts work on directories, not individual files, so this would require any directory containing secrets to be exclusively managed by chezmoi. [rvault](https://github.com/rmind/rvault) seems to have the same idea.

If you want to try building a prototype of this functionality, then the [`chezmoi cat` command](https://www.chezmoi.io/reference/commands/cat/) allows you to ask chezmoi for the target contents of an individual file.

**Reply by infogulch** [reply]
Thanks for the references.

I guess you could create a new directory to bind with FUSE (maybe in `~/.local/` somewhere?), then symlink the file to the FUSE directory. The fuse directory could be considered the root, so the backing fuse impl could just strip off the path prefix to `~/.local/xyz` and pass that path unaltered to `chezmoi cat`, which might simplify the fuse impl. One downside of this is that it could conflict with natural directory permissions, so maybe a different scheme would be better.

Is there a way to get a list of files that need a password manager to generate? It seems a bit silly to do this for .bashrc etc
